common/repo/GitRepo.java

```diff
@@ -150,15 +150,21 @@ public void tagHead(String name) {
   }
 
   public ImmutableList<String> getCommitIds(String branch) {
-    CommandResult commandResult = runCommand("log --pretty=%H master.." + branch);
-    return splitLines(commandResult.stdout);
+    CommandResult commandResult = runCommand("log --pretty=%H master~1.." + branch);
+    // We reverse to return by chronological order
+    return splitLines(commandResult.stdout).reverse();
   }
 
   public ImmutableList<Commit> getCommits(String branch) {
     ImmutableList<String> commits = getCommitIds(branch);
     ImmutableList.Builder<Commit> result = ImmutableList.builder();
     for (String commit : commits) {
-      result.add(Commit.newBuilder().setId(commit).addAllFile(getFilesInCommit(commit)).build());
+      if (commit.equals(commits.get(0))) {
```

Comment:
@oferb you call `commits.get(0)` on each iteration but the result is constant. What's the performance of `.get()` for `ImmutableList`? Just interested, no need to change